hw_05_09/CSVFiles/CSVOperations.py

```diff
@@ -0,0 +1,120 @@
+        fieldnames = ["letter", "count_all", "count_uppercase", "percentage"]  # list of headers
+        try:  # error handler
+            with open(file_path, 'w', newline="", encoding="utf-8") as csv_file:  # open file to write
+                writer = csv.DictWriter(csv_file, fieldnames=fieldnames)  # initialization of the file writer
+                writer.writeheader()  # write headers
+                for sublist in list_of_dicts_with_letters:  # for each provided list item
+                    # write line with provided date
+                    writer.writerow({'letter': sublist[0], 'count_all': sublist[1],
+                                     'count_uppercase': sublist[2], 'percentage': sublist[3]})
+        except BaseException as exception:  # exception handler
+            print(exception)  # print found exception
+
+    @staticmethod
+    def create_csv_files(file_path=FILE_TO_WRITE_PATH):
+        from hw_05_09.OperationsWithFiles.fileReader import FileReader
+        records_list = FileReader.get_records_from_file(file_path)
+        CSVOperations.create_csv_with_words(CSVOperations.get_dict_with_all_words_and_its_count_from_file(records_list))
+        CSVOperations.create_csv_with_letters(CSVOperations.get_dict_of_all_letters_with_its_count_from_file(records_list))
+
+    @staticmethod
+    def get_dict_with_all_words_and_its_count_from_file(records_list: list) -> dict:
+        dict_of_words = {}
+        separated_list_of_words = []
+        for record in records_list:
+            separated_list_of_words.append(re.findall(r"[\w'-]+", record.lower()))  # separating records by words
+        for list_of_words in separated_list_of_words:
+            for word in list_of_words:
+                if word.isalpha() and word not in '\'':
+                    if word in dict_of_words:
+                        dict_of_words[word] += 1
+                    else:
+                        dict_of_words[word] = 1
+        return dict_of_words
+
+    @staticmethod
+    def get_dict_of_all_letters_with_its_count_from_file(records_list):
+        dict_of_letters = {}
+        dict_of_capital_letters = {}
+        separated_list_of_words = []
+        for record in records_list:
+            separated_list_of_words.append(re.split(r'\W+', record))  # separating records to the words
+        for list_of_words in separated_list_of_words:
+            for word in list_of_words:
+                if word.isalpha():
+                    for char in word:
+                        if char.isupper():
+                            if char in dict_of_capital_letters:
+                                dict_of_capital_letters[char] += 1
+                            else:
+                                dict_of_capital_letters[char] = 1
+                        elif char in dict_of_letters:
+                            dict_of_letters[char] += 1
+                        else:
+                            dict_of_letters[char] = 1
```

Comment:
You can get dictionary with all letters:
dict_of_letters[char.lower()] = d_dict_of_letters.get(char.lower(), 0) + 1

Get number of all letters:
 counter_of_all_letters += 1

98-107 rows will be redundant